Adding matrix ``order=None`` to matrix classes

Transurgeon opened the pull request:
This PR adds an optional order argument to some matrix classes.
This is also the first task of issue #492

eriknw commented:
Thanks for starting on this @Transurgeon!

I would recommend adding implementation too. Ideally, you should test and verify what you did is needed and works as expected as you add `order=` arguments. Perhaps this also means adding `my_matrix.orientation` attribute too, or just using the existing `my_matrix.ss.orientation` attribute.

I would start with `A.T.new(order="columnwise")` and test that it works (add tests at the end of `graphblas/tests/test_matrix.py`).

Also, `graphblas/core/utils.py:get_order` will be helpful.

Transurgeon commented:
Hello @eriknw , sorry for the delay on this, I have been finishing up some details for my GSoC project. 
I tried to implement the orientation today, however, it doesn't seem to be working properly. 
I have added the argument to both ``Matrix.__new__`` and ``TransposeMatrix.new()``. I also called ``get_order`` like you recommended, but the test that I have added doesn't seem to pass like I would expect. 
I am starting to get lost in the layers of OOP abstraction. 
More importantly, I am also having problems when debugging due to the error below, do you have any ideas on how to fix this?
<img width="1043" alt="image" src="https://github.com/python-graphblas/python-graphblas/assets/89562186/db82466e-cf62-4b7e-98c8-67ec933f0499">
P.S Is the expectation to add ``property.getter/setter`` for ``.ss.orientation``? I thought that could be a solution but wasn't sure if it was what you wanted.
